src/hotspot/share/cds/dynamicArchive.cpp

```diff
@@ -344,29 +345,34 @@ void DynamicArchive::dump(const char* archive_name, TRAPS) {
   } else {
     // prevent multiple dumps.
     set_has_been_dumped_once();
-  }
-  ArchiveClassesAtExit = archive_name;
-  if (Arguments::init_shared_archive_paths()) {
-    dump();
-  } else {
-    ArchiveClassesAtExit = nullptr;
-    THROW_MSG(vmSymbols::java_lang_RuntimeException(),
+    ArchiveClassesAtExit = archive_name;
+    if (Arguments::init_shared_archive_paths()) {
+      DynamicArchive::dump(CHECK);
```

Comment:
I think you can drop the `DynamicArchive::` in the above since it is called from the same class.

Reply:
> I think you can drop the `DynamicArchive::` in the above since it is called from the same class.

Thanks. Will remove the DynamicArchive:: at push.